### DISCUSSION TITLE
Unused private members for targets in Visual Studio (IDE0051)

### DISCUSSION THREAD

**Discussion by robertcoltheart** [nuke-build/nuke#1468] [Q&A]
Visual studio shows certains targets as being "unused", when clearly they will be used as part of a Nuke build. 

I'm a maintainer over at [Machine.Specifications](https://github.com/machine/machine.specifications) and we have the same issue with using delagates for test targets. 

I'm aware that you can add supressions in code attributes / editorconfig etc, but I'm just wondering if there are any thoughts or ideas as to how you could suppress `IDE0051` at the source instead? I've toyed with the idea of a source analyzer (Roslyn) to suppress the diagnostic or a source generator to "use" the delegates to prevent reporting the diagnostic.

Any thoughts? 

![{0F5DCE11-0EB5-479A-9239-33D99754574F}](https://github.com/user-attachments/assets/ae2d2ef5-c287-4671-8e10-b3ddd59657ed)

**Comment by matkoch** [comment]
Personally, I'm using JetBrains annotations to mark code entities as implicitly used. But this should really only be necessary for targets that are neither a dependency or the default target. I.e., only targets that are manually invoked are a concern. Otherwise, there should always be some kind of usage.
